File: top_interview_questions/python/01_two_sum.py
#!/usr/bin/env python
# encoding: utf-8
'''
@author: Fizz Cao 
@file: 01_two_sum.py
@time: 15/09/2021 9:53 pm
'''
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = dict()

        for i in range(len(nums)):
            if (target - nums[i]) in d:
                return [d[target - nums[i]], i]
            if nums[i] not in d:
                d[nums[i]] = i

File: top_interview_questions/javascript/01_two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var map = new Map();

    for (var i = 0; i < nums.length; i++) {
        if (map.has(target - nums[i]))
            return [map.get(target - nums[i]), i];
        if (!map.has(nums[i]))
            map.set(nums[i], i);
    }
};